core/include/moveit/task_constructor/stage.h

```diff
@@ -179,6 +179,9 @@ class Stage
 	/// marker namespace of solution markers
 	const std::string& markerNS() { return properties().get<std::string>("marker_ns"); }
 
+	void setControllers(std::vector<std::string> controllers) { setProperty("controllers", controllers); }
```

Comment:
I'm not sure it makes sense to offer this API at a stage level since it has nothing to do with planning. I would suggest we add generic trajectory execution properties (a separate map, message or struct) which hold information like this. This way it would also be much easier to add support for additional features and post processing steps like time parameterization or trajectory merging.

Reply:
We want to be able to switch controllers for each and every stage. For example, for closing the grippers, we want to use a force-aware controller to stop closing at a specific grasping force...